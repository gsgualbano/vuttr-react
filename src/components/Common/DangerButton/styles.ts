import styled from 'styled-components';
import { Theme } from '~/styles/global';

interface ButtonProps {
  outline: boolean;
}

type Props = ButtonProps & Theme;

export const Button = styled.button`
  display: inline-flex;
  color: ${({ outline }: ButtonProps) => (outline ? '#f95e5a' : '#ffffff')};
  justify-content: center;
  align-items: center;
  font-size: 16px;
  padding: 5px 10px;
  border: none;
  margin-left: 5px;
  background: ${({ outline, theme }: Props) => (outline ? theme.colors.buttonOutlineBackground : '#f95e5a')};
  border-radius: 5px;

  &:hover {
    ${({ outline }: ButtonProps) => (outline ? 'color: #cc4c4c' : 'background: #cc4c4c')};
  }

  &:click {
    ${({ outline }: ButtonProps) => (outline ? 'color: #a53f3f' : 'background: #a53f3f')};
  }

  @media (max-width: 768px) {
    margin-left: 5px;
    width: 100%;
  }
`;
