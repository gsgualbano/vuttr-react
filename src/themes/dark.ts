export default {
  colors: {
    background: '#1B1A1A',
    textColor: '#fff',
    cardBackground: '#29292B',
    cardBorderColor: '#29292B',
    modalBackground: '#29292B',
    modalBorderColor: '#29292B',
    linkColor: '#ae8fd6',
    linkVisitedColor: '##8050bf',
    buttonOutlineBackground: '#29292B',
    InputBackground: '#9e9fa2',
    InputBorderColor: '#9e9fa2',
  },
};
