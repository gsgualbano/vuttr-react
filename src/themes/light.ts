export default {
  colors: {
    background: '#fcfcfd',
    textColor: '#170c3a',
    cardBackground: '#fcfcfd',
    cardBorderColor: '#ebeaed',
    modalBackground: '#fefefe',
    modalBorderColor: '#888',
    linkColor: 'rgb(0, 0, 238)',
    linkVisitedColor: 'rgb(85, 26, 139);',
    buttonOutlineBackground: '#ffffff',
    InputBackground: '#f5f4f6',
    InputBorderColor: '#ebeaed',
  },
};
