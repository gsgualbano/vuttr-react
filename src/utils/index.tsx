/* eslint-disable max-len */
import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

import { render } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';
import { reducer } from '../store/ducks/tools';

interface Theme {
  colors: {
    background: string;
    textColor: string;
    cardBackground: string;
    cardBorderColor: string;
    modalBackground: string;
    modalBorderColor: string;
    linkColor: string;
    linkVisitedColor: string;
    buttonOutlineBackground: string;
    InputBackground: string;
    InputBorderColor: string;
  };
}

/** redux mock to render components that depends on the redux */
export const renderWithRedux = (
  component: JSX.Element,
  { initialState, store = createStore(reducer, initialState) }: any = {},
) => ({
  ...render(<Provider store={store}>{component}</Provider>),
  store,
});

/** render components thats depends on the theme provider */
export const renderWithTheme = (component: JSX.Element, theme: Theme) => render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

/** render components thats depends on the redux and theme provider */
export const renderWithReduxAndTheme = (
  component: JSX.Element,
  theme: Theme,
  { initialState, store = createStore(reducer, initialState) }: any = {},
) => ({
  ...render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>{component}</Provider>
    </ThemeProvider>,
  ),
  store,
});
