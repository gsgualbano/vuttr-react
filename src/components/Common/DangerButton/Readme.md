#### Botão com a cor vermelho representando danger(perigo)

DangerButton Padrão Exemplo:

```js
<DangerButton>Example</DangerButton>
```

Outline DangerButton Exemplo:

```js
<DangerButton outline>Example</DangerButton>
```

Dark Outline DangerButton Examplo:

```js
import { ThemeProvider } from "styled-components";
import dark from "~/themes/dark";

<ThemeProvider theme={dark}>
  <DangerButton outline>Example</DangerButton>
</ThemeProvider>;
```
