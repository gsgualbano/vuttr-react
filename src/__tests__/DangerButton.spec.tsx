import React from 'react';
import { cleanup } from '@testing-library/react';
import { renderWithTheme } from '~/utils';

import DangerButton from '~/components/Common/DangerButton';

import 'jest-dom/extend-expect';
import 'jest-styled-components';

import dark from '~/themes/dark';
import light from '~/themes/light';

afterEach(cleanup);

describe('<DangerButton />', () => {
  it('render correctly', () => {
    const { asFragment } = renderWithTheme(<DangerButton>Remove</DangerButton>, light);
    expect(asFragment()).toMatchSnapshot();
  });

  it('have correct color', () => {
    const { getByTestId } = renderWithTheme(<DangerButton>Remove</DangerButton>, light);
    expect(getByTestId('danger-button')).toHaveStyleRule('background', '#f95e5a');
    expect(getByTestId('danger-button')).toHaveStyleRule('color', '#ffffff');
    expect(getByTestId('danger-button')).toHaveStyleRule('background', '#cc4c4c', {
      modifier: ':hover',
    });
  });

  it('have correct outline color', () => {
    const { getByTestId } = renderWithTheme(<DangerButton outline>Remove</DangerButton>, light);
    expect(getByTestId('danger-button')).toHaveStyleRule('background', '#ffffff');
    expect(getByTestId('danger-button')).toHaveStyleRule('color', '#f95e5a');
    expect(getByTestId('danger-button')).toHaveStyleRule('color', '#cc4c4c', {
      modifier: ':hover',
    });
  });

  it('have correct dark outline color', () => {
    const { getByTestId } = renderWithTheme(<DangerButton outline>Remove</DangerButton>, dark);
    expect(getByTestId('danger-button')).toHaveStyleRule('background', '#29292B');
    expect(getByTestId('danger-button')).toHaveStyleRule('color', '#f95e5a');
    expect(getByTestId('danger-button')).toHaveStyleRule('color', '#cc4c4c', {
      modifier: ':hover',
    });
  });
});
