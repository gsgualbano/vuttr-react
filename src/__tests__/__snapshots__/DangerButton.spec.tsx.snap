// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DangerButton /> render correctly 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #ffffff;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  padding: 5px 10px;
  border: none;
  margin-left: 5px;
  background: #f95e5a;
  border-radius: 5px;
}

.c0:hover {
  background: #cc4c4c;
}

.c0:click {
  background: #a53f3f;
}

@media (max-width:768px) {
  .c0 {
    margin-left: 5px;
    width: 100%;
  }
}

<button
    class="c0"
    data-testid="danger-button"
    type="button"
  >
    Remove
  </button>
</DocumentFragment>
`;
